lib/awful/widget/calendar.lua

```diff
@@ -0,0 +1,381 @@
+---------------------------------------------------------------------------
+-- A calendar popup wibox
+--
+-- Display a month or year calendar popup using `calendar.month` or `calendar.year`.
+-- The calendar style can be tweaked by providing tables of style properties at creation:
+-- `style_year`, `style_month`, `style_yearheader`, `style_header`,
+-- `style_weekdays`, `style_weeknumber`, `style_normal`, `style_focus` (see `cell_properties`).
+--
+-- The wibox accepts arguments for the calendar widget: "font", "spacing", "week_numbers", "start_sunday".
+-- It also accepts the extra arguments `opacity`, `bg`, `screen` and `position`.
+-- `opacity` and `bg` apply to the wibox itself, they are mainly useful to manage opacity
+-- by setting `opacity` for the false opacity or setting `bg="#00000000"` for compositor opacity.
+-- The `screen` argument forces the display of the wibox to this screen (instead of the focused screen by default).
+-- The `position` argument is a two-characters string describing the screen alignment "[vertical][horizontal]",
+-- e.g. "cc", "tr", "bl", ...
+--
+-- The wibox visibility can be changed calling the `toggle` method.
+-- The `attach` method adds mouse bindings to an existing widget in order to toggle the display of the wibox.
+--
+--@DOC_wibox_awidget_defaults_calendar_EXAMPLE@
+--
+-- @author getzze
+-- @copyright 2017 getzze
+-- @classmod awful.widget.calendar
+---------------------------------------------------------------------------
+
+local setmetatable = setmetatable
+local string = string
+local gtable = require("gears.table")
```

Comment:
You already import gears below, you could remove this line and expand `gtable` to `gears.table`